net-dialup/xl2tpd/xl2tpd-1.3.11-r1.ebuild

```diff
@@ -0,0 +1,57 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit systemd toolchain-funcs
+
+DESCRIPTION="A modern version of the Layer 2 Tunneling Protocol (L2TP) daemon"
+HOMEPAGE="http://www.xelerance.com/services/software/xl2tpd/"
+SRC_URI="https://github.com/xelerance/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~mips ~ppc ~ppc64 ~x86"
+IUSE="dnsretry"
+
+DEPEND="
+	net-libs/libpcap
+	>sys-kernel/linux-headers-2.6"
+
+RDEPEND="
+	${DEPEND}
+	net-dialup/ppp"
+
+DOCS=(CREDITS README.xl2tpd BUGS CHANGES TODO doc/README.patents)
+
+PATCHES=( "${FILESDIR}/${P}-ENODEV-fix.patch" )
+
+src_prepare() {
+	default
+	# The below patch is questionable. Why wasn't it submitted upstream? If it
+	# ever breaks, it will just be removed. -darkside 20120914
+	# Patch has been discussed with upstream and is marked as feature by now:
+	# https://github.com/xelerance/xl2tpd/issues/134 // -- tenX 2017-08-06
+	use dnsretry && eapply -p0 "${FILESDIR}/${PN}-dnsretry.patch"
```

Comment:
why conditional patchung?

Reply:
Not added by me...it's in all the other versions, these are just version bumps. However, if you read the upstream issue this is a Gentoo-specific patch which changes upstream behavior.